tutorials/fileIO/src/test/java/com/example/fileIO/ThroughputBasedApplicationTest.java

```diff
@@ -0,0 +1,124 @@
+  {
+    for (int file = 0; file < nFiles; file++) {
+      ArrayList<String> lines = new ArrayList<>();
+      for (int line = 0; line < nLines; line++) {
+        lines.add("file " + file + ", line " + line);
+      }
+      try {
+        FileUtils.write(new File(inputDirFile, "file" + file), StringUtils.join(lines, "\n"));
+      } catch (IOException e) {
+        System.out.format("Error: Failed to create file %s%n", file);
+        e.printStackTrace();
+      }
+    }
+    System.out.format("Created %d files with %d lines in each%n", nFiles, nLines);
+  }
+
+  private void cleanup()
+  {
+    try {
+      FileUtils.deleteDirectory(inputDirFile);
+      FileUtils.deleteDirectory(outputDirFile);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  // check that the requisite number of files exist in the output directory
+  private boolean checkFilesCopied(final int nFiles)
+  {
+    return (nFiles == FileUtils.listFiles(outputDirFile, null, false).size())
+        && (new File(outputDirFile, "file0").exists());
+  }
+
+  // return Configuration with suitable properties set
+  private Configuration getConfig()
+  {
+    final Configuration result = new Configuration(false);
+    result.setInt("dt.application.ThroughputBasedFileIO.attr.CHECKPOINT_WINDOW_COUNT", 10);
+    result.set("dt.application.ThroughputBasedFileIO.operator.read.prop.directory", inputDirName);
+    result.setInt("dt.application.ThroughputBasedFileIO.operator.read.prop.pendingFilesPerOperator", 2);
+    result.setInt("dt.application.ThroughputBasedFileIO.operator.read.prop.partitionCount", numPartitions);
+    result.set("dt.application.ThroughputBasedFileIO.operator.write.prop.filePath", outputDirName);
+    return result;
+  }
+
+  @Before
+  public void beforeTest() throws Exception
+  {
+    cleanup();
+    FileUtils.forceMkdir(inputDirFile);
+    FileUtils.forceMkdir(outputDirFile);
+
+    // create some text files in input directory
+    createFiles(numFiles, numLines);
+  }
+
+  @After
+  public void afterTest()
+  {
+    cleanup();
+  }
+
+  @Test
+  public void testApplication() throws Exception
+  {
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      lma.prepareDAG(new ThroughputBasedApplication(), getConfig());
+      LocalMode.Controller lc = lma.getController();
+      lc.runAsync();
+
+      // wait for output files to show up
+      while (!checkFilesCopied(numFiles)) {
+        System.out.println("Waiting for files to get copied.");
+        Thread.sleep(1000);
+      }
+      verifyCopiedData();
+    } catch (ConstraintViolationException e) {
+      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    }
+  }
+
+  private void verifyCopiedData() throws IOException
+  {
+    for (int i = 0; i < numFiles; i++) {
+      String fileName = "file" + i;
+      String inputFileData = FileUtils.readFileToString(new File(inputDirFile, fileName)).trim();
```

Comment:
Could you please check what is the difference between two files? is it because of newline?